src/napari/_qt/widgets/qt_welcome.py

```diff
@@ -188,15 +186,19 @@ def dropEvent(self, event):
             Event from the Qt context.
         """
         self._update_property('drag', False)
-        self.sig_dropped.emit(event)
+        if (
+            self.parent() is not None
```

Comment:
Could be helpful to comment on why we recurse 3 levels.

Reply:
We should remove this code in #8117

It comes from internal structure.

QtViewer -> Widget -> Stacked Widgets -> This widget